### ISSUE TITLE
Buttons opening dropdown should better indicate its content with aria-haspopup property

### ISSUE THREAD

**Issue by mlewand** [ckeditor/ckeditor-dev#2072]
## Type of report

Feature request

## Provide description of the new feature

WAI-ARIA 1.1 has added new values for [`aria-haspopup`](https://www.w3.org/TR/wai-aria-1.1/#aria-haspopup) property, among them are `listbox` and `menu`. This allows us to better indicate what is contained by button that has a popup.

We have some buttons that opens `[role=menu]` elements (e.g. `menubutton`) and several other that work on `listbox`es (e.g. style combo, color button).

Currently their buttons `aria-haspopup` is set to `true`. It should get more precise value for each case.

**Comment by mlewand** [comment]
One important thing is to ensure that our official a11y environment works correctly with this change. We don't want to break a11y support in case if JAWS did not caught up with WAI-ARIA 1.1 and treats `[aria-haspopup=listbox]` as `[aria-haspopup=false]`.

**Comment by Comandeer** [comment]
Merged into `next` in c5c5e39.

**Comment by mlewand** [comment]
Note that this change can already be previewed on our [nightly preview demo](https://ckeditor4-preview.ckeditor.com/major/samples/index.html).
